tmt/export.py

```diff
@@ -130,16 +143,17 @@ def convert_manual_to_nitrate(test_md):
                 sections_headings[key] = result
                 break
 
-    def concatenate_headings_content(headings):
+    def concatenate_headings_content(headings: Tuple[str, ...]) -> HeadingsType:
         content = list()
         for v in headings:
             content += sections_headings[v]
         return content
 
-    def enumerate_content(content):
-        content.sort()
+    def enumerate_content(content: HeadingsType) -> HeadingsType:
+        # for sorting convert the index to integer, but keep whole list as list of strings
+        content.sort(key=lambda a: int(a[0]))
         for c in range(len(content)):
-            content[c][1] = f"<p>Step {c + 1}.</p>" + content[c][1]
+            content[c][1] = f"<p>Step {c + 1}.</p>" + cast(str, content[c][1])
```

Comment:
`content` is `HeadingsType`, i.e. `List[List[Union[int, str]]]`, `content[c]` is therefore `List[Union[int, str]]`, and `content[c][1]` would be `Union[int, str]` - `cast()` does not seem to be safe enough, it would not convert `int` to `str`, it'd only convince mypy to treat the value as string while it still may be an integer. If it cannot be an integer, `assert isinstance(content[c][1], str)` would be better, to be absolutely sure we're adding `str` and `str`, not `str` + `int` as allowed by the type.

Reply:
it is a mess :( will try something